scripts/addr2line.py

```diff
@@ -25,11 +25,18 @@
 import subprocess
 from enum import Enum
 from functools import cache
-from typing import Any
+from typing import Any, Optional, TypeVar, Union, cast
 
 # special binary path/module indicating that the address is from the kernel
 KERNEL_MODULE = '<kernel>'
 
+T = TypeVar('T')
+
```

Comment:
Agreed, and thanks for all of these comments. In fact, I think it would make sense to sprinkle some auto-format magic & enforcement on this project, WDYT? I find auto-format works well in Python, and it seems like `black` aligns quite closely with the existing format (and most of the deltas are just inconsistencies in the existing formatting).

In that spirit I submitted https://github.com/scylladb/seastar/pull/2441 which uses black as the formatting for these 2 addr2line files and enforces it in CI. If that goes through I'll rebase these changes on top of that and reformat which should resolve your formatting related concerns.